spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingProperties.java

```diff
@@ -33,7 +35,17 @@
 @JsonInclude(Include.NON_DEFAULT)
 public class ChannelBindingProperties {
 
-	public static final String PATH = "path";
+	public static final String DESTINATION = "destination";
+
+	public static final String PARTITION_COUNT = "partitionCount";
+
+	public static final String PARTITIONED = "partitioned";
+
+	@Value("${INSTANCE_INDEX:${CF_INSTANCE_INDEX:0}}")
+	private int instanceIndex;
+
+	@Value("${INSTANCE_COUNT:1}")
```

Comment:
Given `INSTANCE_INDEX` and `CF_INSTANCE_INDEX` would be available in lattice/CF environments, not sure the reasoning behind adding `INSTANCE_COUNT` as an environment variable here.